components/marketplace/list-for-sale-button.tsx

```diff
@@ -19,7 +19,13 @@ import { getAddress } from "viem";
 
 import { isChainIdSupported } from "@/lib/isChainIdSupported";
 
-export function ListForSaleButton({ hypercert }: { hypercert: HypercertFull }) {
+export function ListForSaleButton({
+  hypercert,
+  disabledForChain = false,
+}: {
+  hypercert: HypercertFull;
+  disabledForChain?: boolean;
```

Comment:
I feel like there is an argument to be made for doing this inside the component, instead of passed as a prop. Keeps the interface of the component cleaner, and much harder to forget when adding this button somewhere else.

Reply:
It all comes down to one question: Do we want this button and the others like it to express this chain-specific behavior or is there a case where we wouldn't disable this on any chain if we do reuse it?

Reply:
It follow the pattern from the Vercel/NextJS SDK and I can see their arguments: https://flags-sdk.dev/knowledge-base/server-side-vs-client-side